pkg/build/builder/sti.go

```diff
@@ -40,23 +41,37 @@ func NewSTIBuilder(client DockerClient, dockerSocket string, authCfg docker.Auth
 
 // Build executes the STI build
 func (s *STIBuilder) Build() error {
-	tag := s.build.Parameters.Output.DockerImageReference
+	var push bool
+
+	// if there is no output target, set one up so the docker build logic
+	// will still work, but we won't push it at the end.
+	if s.build.Spec.Output.To == nil || len(s.build.Spec.Output.To.Name) == 0 {
+		s.build.Spec.Output.To = &kapi.ObjectReference{
+			Kind: "DockerImage",
+			Name: noOutputDefaultTag,
+		}
+		push = false
+	} else {
+		push = true
```

Comment:
i've got to setup the no outputtag anyway, so i can't reverse the if and nuke the else...?